---
layout: default
---
# BioC 2019: Where Software and Biology Connect

When: July 25 (Developer Day) 26 and 27, 2018 <br />
Where: [Victoria University][venue], University of Toronto, Toronto, Canada<br />
Twitter: [#bioc2019][tweet]

[tweet]: https://twitter.com/hashtag/bioc2019?f=tweets
[venue]: ./travel-accommodations

## Schedule

* [Developer Day](schedule-developer-day): Wednesday, July 25.
* [Day 1](#day-1-thursday-july-26): Thursday July 26.
* [Day 2](#day-2-friday-july-27): Friday July 27.

[1]: http://sites.utoronto.ca/andrewslab/
[2]: https://www.pmgenomics.ca/bhklab/
[3]: https://www.rits.onc.jhmi.edu/DBB/members/?members=Faculty&member=efertig1
[4]: https://csoneson.github.io/
[5]: https://hoffmanlab.org/
[6]: http://hugheslab.med.utoronto.ca/

### Developer Day: Wednesday July 25

See the [Developer Day](schedule-developer-day) schedule.
   
### Day 1: Thursday July 26

Logistics:

- Start your [course AMI][]
- Join the [bioc-community slack][]

[course AMI]: https://courses.bioconductor.org
[bioc-community slack]: https://bioc-community.herokuapp.com/

8:00 - 8:45 -- Registration and breakfast, VC second floor foyer
: 

8:45 - 9:00 -- Welcoming remarks -- [slides][1.0] (Martin Morgan)
: VC 213 Chapel

9:00 - 9:30 -- [Michael Hoffman][5] University of Toronto.
: Virtual ChIP-seq: predicting transcription factor binding by
  learning from the transcriptome -- VC 213 Chapel

9:30 - 10:00 -- [Elana Fertig][3] John Hopkins University. 
: Enter the Matrix: Interpreting omics through matrix factorization -- VC 213 Chapel

10:00 - 10:30 -- Break
: 

10:30 - 11:00 [Brenda Andrews][1] University of Toronto.
: Analysis of high content microscopy data generated through automated
  yeast genetics -- VC 213 Chapel

11:00 - 12:00 -- Contributed talks -- VC 213 Chapel
: + Innes BT\* and Bader GD. scClustViz - Single-cell RNAseq Cluster
    Assessment and Interactive Visualisation.
  + Righelli D\*, Koberstein J, Gomes B, Zhang N, Angelini C,
    Peixoto L, Risso D. Differential Enriched Scan 2 (DEScan2): a
    fast pipeline for broad peak analysis.
  + Adithya M, Bhargava A, Wright E\*. Improving the accuracy of
    taxonomic classification for identifying taxa in microbiome
    samples.
  + Zhun Miao\*, Ke Deng, Xiaowo Wang, Xuegong Zhang. DEsingle for
    detecting three types of differential expression in single-cell
    RNA-seq data.
  + Ludwig Geistlinger\*, Gergely Csaba, Mara Santarelli, Lucas
    Schiffer, Marcel Ramos, Ralf Zimmer, and Levi Waldron. Towards a
    gold standard for benchmarking gene set enrichment analysis.

12:00 - 1:00 -- Lunch / Birds-of-a-feather -- VC Foyer and Alumni Hall
: 
  + Organizing and coordinating R / Bioconductor meetups -- VC 211

1:00 - 2:45 --  Workshop Session 1a
: + Martin Morgan\*. [Bioconductor for Everyone][100] -- VC 212
  + Ludwig Geistlinger\* and Levi Waldron. [Functional enrichment
    analysis of high-throughput omics data][210] -- VC 215

1:45 - 2:45 --  Workshops Session 1b
: + Zhaleh Safikhani\*, Petr Smirnov, Benjamin
    Haibe-Kains. [Biomarker discovery from large pharmacogenomics
    datasets][260] -- VC 206

2:45 - 3:15 -- Break
: 

3:15 - 5:00 --  Workshops Session 2a
: + MacDonald J, Shepherd L. [Introduction to Bioconductor annotation
    resources][101] -- VC 206
  + Love MI. [RNA-seq data analysis with DESeq2][201] -- VC 215

4:00 - 5:00 --  Workshops Session 2b
: + <del>Coetzee SG\* and Hazelett DJ. Variant Functional Annotation
    using StatePaintR, FunciVar and MotifBreakR</del>
  + Isserlin R\*, Innes B, Bader
    GD. [Cytoscape Automation in R using Rcy3][230] -- **SET UP** your
    laptop for this workshop by arriving at 3:30. For the workshop
    [download and install cytoscape 3.6.1][cyto] and [workshop files][cytowork] -- VC 212

5:30 - 7:30 -- Contributed posters -- VC Foyer and Alumni Hall
: 

[1.0]: https://docs.google.com/presentation/d/1W0tzadEgIKSq6J5Bcq7iBgcEIgRlSlbdtMRpDaRQ9m8/edit?usp=sharing
[cyto]: http://cytoscape.org/download.php
[cytowork]: http://download.baderlab.org/Bioc2019/

### Day 2: Friday July 27

8:00 - 8:30 -- Breakfast, VC Second floor foyer

8:30 - 9:00 -- [Tim Hughes][6], University of Toronto.
: Binding motifs for DNA and RNA binding proteins -- VC 213 Chapel

9:00 - 9:30 -- [Benjamin Haibe-Kains][2], Princesss Margaret Cancer
  Center, Toronto.
: Cancer Biomarker Discovery: Building a Bridge Between Preclinical
  and Clinical Research -- VC 213 Chapel

9:30 - 10:00 --  Contributed talks -- VC 213 Chapel
: + Lee S\*, Cook D, Lawrence M. plyranges: a fluent interface to
    Bioconductor's Ranges infrastructure
  + Love MI\*, Hickey P, Soneson, C, and Patro R. Automatic metadata
    propagation for RNA-seq


10:00 - 10:30 -- Break
: 

10:30 - 11:00 -- [Charlotte Soneson][4], University of Zurich, Switzerland. 
: A junction coverage compatibility score to quantify the reliability
  of transcript abundance estimates and annotation catalogs  -- VC 213 Chapel

11:00 - 12:00 -- Contributed talks  -- VC 213 Chapel
: + Albert Y Zhang, Shian Su, Matthew E Ritchie, and Charity W
    Law\*. Unpacking signal from RNA-seq intron reads using Rsubread
    and limma packages.
  + Steinbaugh MJ\*, Kirchner RD, Ho Sui S. bcbioSingleCell: R
    package for bcbio single-cell analysis.
  + Abbas Rizvi\*, Ezgi Karaesmen\*, Leah Preus, Michael Sovic,
    Junke Wang, Lara Sucheston-Campbell. gwasurvivr: an R package to
    perform survival association testing on imputed genetic data.
  + Nima Hejazi\*, Alan Hubbard, Mark van der Laan. Data-Adaptive
    Estimation and Inference for Differential Methylation Analysis.
  + Rachael V Phillips\*, Alan Hubbard. Data Adaptive Evaluation of
    Preprocessing Methods using Ensemble Machine Learning.

12:00 - 1:00 -- Lunch / Birds-of-a-feather -- VC Foyer and Alumni Hall
: 

1:00 - 2:45 -- Workshops Session 3a
: + Michael Lawrence\*, Martin Morgan. [Solving common bioinformatic
    challenges using GenomicRanges][102] -- VC 215
  + Ramos M, Geistlinger L, Waldron L\*. [Workflow for Multi-omics
    Analysis with MultiAssayExperiment][220] -- VC 212

1:45 - 2:45 -- Workshops Session 3b
: + Charity Law\* , Monther Alhamdoosh, Shian Su, Gordon Smyth and
    Matthew Ritchie. [RNA-seq analysis is easy as 1-2-3 with limma,
    Glimma and edgeR][200] -- VC 206

2:45 - 3:15 -- Break
: 

3:15 - 5:00 -- Workshops Session 4a
: + Das D\*, Street K\*, Risso D\*. [Analysis of single-cell RNA-seq
    data: Dimensionality reduction, clustering, and lineage
    inference][202] -- VC 215
  + Nicholas Cooley\*, Erik Wright. [Working with Genomic Data in R
    with the DECIPHER package][250] -- VC 212
  + Levi Waldron, Benjamin Haibe-Kains, Sean Davis. [Public Data
    Resources and Bioconductor][103] -- VC 211

4:00 - 5:00 -- Workshops Session 4b
: + Lee S\*, Lawrence M. [Fluent genomic data analysis with plyranges][240]
    -- VC 206

5:30 - 7:30 -- Contributed posters -- VC Foyer and Alumni Hall
: 


[100]: http://bioconductor.github.io/BiocWorkshops/r-and-bioconductor-for-everyone-an-introduction.html
[101]: http://bioconductor.github.io/BiocWorkshops/introduction-to-bioconductor-annotation-resources.html
[102]: http://bioconductor.github.io/BiocWorkshops/solving-common-bioinformatic-challenges-using-genomicranges.html
[103]: http://bioconductor.github.io/BiocWorkshops/public-data-resources-and-bioconductor.html

[200]: http://bioconductor.github.io/BiocWorkshops/rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html
[201]: http://bioconductor.github.io/BiocWorkshops/rna-seq-data-analysis-with-deseq2.html
[202]: http://bioconductor.github.io/BiocWorkshops/analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html
[210]: http://bioconductor.github.io/BiocWorkshops/functional-enrichment-analysis-of-high-throughput-omics-data.html
[220]: http://bioconductor.github.io/BiocWorkshops/workflow-for-multi-omics-analysis-with-multiassayexperiment.html
[230]: http://bioconductor.github.io/BiocWorkshops/cytoscape-automation-in-r-using-rcy3.html
[240]: http://bioconductor.github.io/BiocWorkshops/fluent-genomic-data-analysis-with-plyranges.html
[250]: http://bioconductor.github.io/BiocWorkshops/working-with-genomic-data-in-r-with-the-decipher-package.html
[260]: http://bioconductor.github.io/BiocWorkshops/biomarker-discovery-from-large-pharmacogenomics-datasets.html
